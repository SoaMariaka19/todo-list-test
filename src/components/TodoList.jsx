import React, { useState } from "react";
import TodoMap from "./TodoMap";
import DoneMap from "./DoneMap";

function TodoList() {
  const [todoList, setTodoList] = useState([]);
  const [doneList, setDoneList] = useState([]);
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleAddTodo = () => {
    setTodoList([...todoList, inputValue]);
    setInputValue("");
  };

  const handleCheckboxChange = (index) => {
    const newTodoList = [...todoList];
    setDoneList([...doneList, newTodoList[index]]);
    newTodoList.splice(index, 1);
    setTodoList(newTodoList);
  };

  const handleRemoveDone = (index) => {
    const newDoneList = [...doneList];
    newDoneList.splice(index, 1);
    setDoneList(newDoneList);
  };

  return (
    <div>
    <form onSubmit={(e)=>e.preventDefault()}>
      <input
        type="text"
        value={inputValue}
        placeholder='Enter Todo'
        onChange={handleInputChange}
      />
      <button
          data-testid='btn'
          className="btn-todo" onClick={handleAddTodo}>Add Todo</button>
    </form>
      <h3>Todo:</h3>
      <ul
          data-testid="ulTodo">
        <TodoMap todoList={todoList} handleCheckboxChange={handleCheckboxChange} />
      </ul>
      <h3>Done:</h3>
      <ul
          data-testid="ulDone">
        <DoneMap doneList={doneList} handleRemoveDone={handleRemoveDone} />
      </ul>
    </div>
  );
}

export default TodoList;
