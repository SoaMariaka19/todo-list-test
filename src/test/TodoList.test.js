import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent ,screen} from "@testing-library/react";
import TodoList from "../components/TodoList";

test("render todolist component" , () => {
  const tree = renderer.create(<TodoList/>);
  const todolist_component = tree.toJSON();
  expect(todolist_component).toMatchSnapshot();
})


test("should return input" , ()=>{
  const testing_compo =  render(<TodoList/>);
  const input = testing_compo.getByPlaceholderText("Enter Todo");
  expect(input).toMatchSnapshot();
})


test("should return button" , ()=>{
  const testing_compo =  render(<TodoList/>);
  const button = testing_compo.getAllByTestId("btn");
  expect(button).toMatchSnapshot();
})


test("h3 context todo" , ()=>{
  render(<TodoList/>);
  const h3 = screen.getByText("Todo:");
  expect(h3).toBeInTheDocument();
})


test("h3 context done" , ()=>{
  render(<TodoList/>);
  const h3 = screen.getByText("Done:");
  expect(h3).toBeInTheDocument();
})


test("inputValue updates when text is entered", () => {
  const { getByPlaceholderText } = render(<TodoList />);
  const input = getByPlaceholderText("Enter Todo");
  fireEvent.change(input, { target: { value: "Buy Milk" } });
  expect(input.value).toBe("Buy Milk");
});


test('Adding a new item to the todoList when the user clicks the "Add Todo" button', () => {
  const { getByPlaceholderText, getByText } = render(<TodoList />);
  const input = getByPlaceholderText('Enter Todo');
  const addTodoButton = getByText('Add Todo');

  fireEvent.change(input, { target: { value: 'Test Todo' } });
  fireEvent.click(addTodoButton);

  const todoList = getByText('Test Todo');
  expect(todoList).toBeInTheDocument();
});


it('should render without crashing', () => {
  render(<TodoList />);
});





