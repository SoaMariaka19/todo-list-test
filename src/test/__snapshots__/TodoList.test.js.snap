// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render todolist component 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Enter Todo"
      type="text"
      value=""
    />
    <button
      className="btn-todo"
      data-testid="btn"
      onClick={[Function]}
    >
      Add Todo
    </button>
  </form>
  <h3>
    Todo:
  </h3>
  <ul
    data-testid="ulTodo"
  />
  <h3>
    Done:
  </h3>
  <ul
    data-testid="ulDone"
  />
</div>
`;

exports[`should return button 1`] = `
Array [
  <button
    class="btn-todo"
    data-testid="btn"
  >
    Add Todo
  </button>,
]
`;

exports[`should return input 1`] = `
<input
  placeholder="Enter Todo"
  type="text"
  value=""
/>
`;
